server/handler/details.go

```diff
@@ -121,31 +120,33 @@ func (h *Details) ServeHTTP(w http.ResponseWriter, r *http.Request) error {
 
 	config, err := h.ConfigFetcher.ConfigForPR(ctx, prctx, client)
 	data.PolicyURL = getPolicyURL(pr, config)
-
 	if err != nil {
-		data.Error = errors.WithMessage(err, fmt.Sprintf("Failed to fetch configuration at ref=%s", config.Ref))
+		data.Error = errors.WithMessage(err, fmt.Sprintf("Failed to fetch configuration: %s", config))
 		return h.render(w, data)
 	}
 
-	if config.Missing() {
-		data.Error = errors.New(config.Description())
-		return h.render(w, data)
-	}
+	evaluator, err := h.Base.ValidateFetchedConfig(ctx, prctx, client, config, common.TriggerAll)
 
-	if config.Invalid() {
-		data.Error = errors.WithMessage(config.Error, config.Description())
+	if err != nil {
+		data.Error = err
 		return h.render(w, data)
 	}
 
-	evaluator, err := policy.ParsePolicy(config.Config)
-	if err != nil {
-		data.Error = errors.WithMessage(err, fmt.Sprintf("invalid policy at ref \"%s\"", config.Ref))
+	if evaluator == nil {
+		data.Error = errors.Errorf("Unable to evaluate: %s", config.Description())
```

Comment:
From your other comment, `evalutator` is nil when the policy is not defined, right? Maybe this message should be something like `No policy defined for %s`

Reply:
The evaluator is `nil` without an error when config is missing or the trigger does not match. You're right, when `TriggerAll` is passed in that should only be the case for missing configs. Adding a specific formatted error like that seems a little tightly bound, although it is a lot simpler. Right now the error should resolve to:

```Unable to evaluate: No policy found at <owner>/<repo> ref:<ref>.```

For example, if a status trigger was introduced without being added to `TriggerAll`, this hardcoded error might lead the user down the wrong debugging path. That might be a very specific case though — what do you think?

Perhaps the error ```Unable to evaluate nil evaluator: ...``` would be a better message to add here.

The reason this error needs to be created here is that it is ignored elsewhere. That is a little weird, and perhaps a better way of going about this would be to emit an error for missing and non-triggered evaluations, then catch and suppress that error in the callers which prefer to ignore them. This would be more verbose, but much more explicit. I could make the errors `IgnoreableErrors`'s in the same way we added `TemporaryError`'s elsewhere. Maybe that is overkill — what do you think?

Reply:
The original message is fine as-is; I didn't realize that the `config.Description()` included the information about the policy being missing so I thought we were confusing invalid policies with missing policies.

I don't like the idea of a generally ignorable error. If we needed this, I think we'd want a specific error code/type that indicates the failure condition. But as I mentioned, I don't think we actually need to change anything.